redux/selectors.ts

```diff
@@ -3,6 +3,15 @@ import { RootState } from "./store";
 import data from "../data";
 import { VALUE, QUANTITY, INGREDIENT } from "./reducers/recipes";
 
+export const selectUserNutriscore = createSelector(
+  (state: RootState, currentRecipeId: string) => currentRecipeId,
+  (state: RootState) => state.recipe.recipes,
+  (currentRecipeId, recipes) => recipes[currentRecipeId].nutriscore
+);
+
+//   (state: RootState) => state.recipe.recipes["userRecipe"].nutriscore
```

Comment:
Because create selector is more complex than expected for optimization purpose.

all the functions except the last one are run and the result is saved.

The last one take the saved result of the previous one and the selector returns its value

It's useful because if you update the editor state, the recipes will not be touched sot the first function will return the same value. Then the selector stops its execution since it already now the result will be the same (a bit overkill for our application)

You could do it  in two lines
```
 (state: RootState, recipeId: string) => state.recipe.recipes[recipeId].nutriscore,
  (nutriscore) => nutriscore,
```

Or simpler: don't use selector at all

```js
const userNutriscore = store.getState().recipe.recipes.userRecipe.nutriscore
```

Reply:
That's much simpler, thanks!